### PR TITLE
Add a changelog file for dev

### PR DESCRIPTION
**Pull request by namurphy**
This is a stub file for the towncrier build.

### CODE REVIEW

File: docs/whatsnew/0.9.0.rst [hunk 1]
```diff
@@ -0,0 +1,5 @@
+Version v0.9.0 (TBD)
+====================
+
```
Comment (rocco8773) [inline docs/whatsnew/0.9.0.rst:3]:
```suggestion
```

`towncrier` should auto-generated the title based on the toml configuration....At least, that's how we had it setup in the past.

**Comment by namurphy** [comment]
I'm getting some errors.  Still need to figure out what's causing them...
```
/home/runner/work/PlasmaPy/PlasmaPy/docs/whatsnew/0.9.0.rst:1: CRITICAL: string indices must be integers
/home/runner/work/PlasmaPy/PlasmaPy/docs/whatsnew/index.rst:11: WARNING: toctree contains reference to document 'whatsnew/0.9.0' that doesn't have a title: no link will be generated
```

File: pyproject.toml [hunk 2]
```diff
@@ -36,36 +36,33 @@ lines_between_types = 1
...
-[[tool.towncrier.type]]
-directory = "doc"
-name = "Improved Documentation"
+[tool.towncrier.type.removal]
+name = "Deprecations and Removals"
 showcontent = true
 
-[[tool.towncrier.type]]
-directory = "trivial"
+[tool.towncrier.type.trivial]
```
Comment (rocco8773) [inline pyproject.toml:66]: I've been running into some weird stuff with this new configuration.  It's just buggy and hard to explain the issues, but in short... This new configuration syntax was [added back in Feb 24, 2022](https://github.com/twisted/towncrier/commit/bda743b945f8cbaed088a9ba8cceacc9d1c98a79), so it's not even available in the current [21.9.0 release (Feb. 10, 2022)](https://pypi.org/project/towncrier/#history).  As such, `sphinx_changelog` (which we use to generate the dev changelog) has not incorporated the current version of `towncrier`.

TL;DR We need to put this back to the old style configuration to avoid `towncrier` bugs.

**Comment by rocco8773** [comment]
> I'm getting some errors. Still need to figure out what's causing them...
> 
> ```
> /home/runner/work/PlasmaPy/PlasmaPy/docs/whatsnew/0.9.0.rst:1: CRITICAL: string indices must be integers
> /home/runner/work/PlasmaPy/PlasmaPy/docs/whatsnew/index.rst:11: WARNING: toctree contains reference to document 'whatsnew/0.9.0' that doesn't have a title: no link will be generated
> ```

See my above comment.  This is part of the bugs associated with the toml configuration.

File: pyproject.toml [hunk 3]
```diff
@@ -36,36 +36,33 @@ lines_between_types = 1
 
 [tool.towncrier]
 package = "plasmapy"
+name = "PlasmaPy"
 filename = "CHANGELOG.rst"
 directory = "changelog/"
-issue_format = "`#{issue} <https://github.com/plasmapy/plasmapy/pull/{issue}>`__"
+wrap = true
```
Comment (rocco8773) [inline pyproject.toml:42]: What is this?  I don't see it anywhere in the `towncrier` documentation?

File: pyproject.toml (hunk 2 above)
Comment (namurphy) [reply pyproject.toml:66]: Sigh!

**Comment by rocco8773** [comment]
I did a local build still keep this part of the configuration and it worked.

```toml
[tool.towncrier]
package = "plasmapy"
name = "PlasmaPy"
filename = "CHANGELOG.rst"
directory = "changelog/"
title_format = "{name} v{version} ({project_date})"
issue_format = ":issue:`{issue}`"
```

File: docs/whatsnew/0.9.0.rst [hunk 4]
```diff
@@ -0,0 +1,2 @@
+.. changelog::
+   :towncrier: ../../
```
Comment (rocco8773) [inline docs/whatsnew/0.9.0.rst:2]:
```suggestion
.. changelog::
   :towncrier: ../../
   :towncrier-skip-if-empty:
```

If we use the [`:towncrier-skip-if-empty:` option](https://sphinx-changelog.readthedocs.io/en/latest/#skipping-towncrier-on-release-builds), then we could just name this file `dev.rst` and not have to worry about renaming/deleting it during releases.

File: pyproject.toml (hunk 3 above)
Comment (namurphy) [reply pyproject.toml]: I found it under [further options](https://towncrier.readthedocs.io/en/latest/index.html?highlight=wrap#further-options), to limit lines to 79 characters.

I'm giving it a try in #1626 to see if it works.

File: docs/whatsnew/0.9.0.rst (hunk 4 above)
Comment (namurphy) [reply docs/whatsnew/0.9.0.rst:2]: Cool!

**Comment by namurphy** [comment]
I had originally added the changelog entry to test it with $> 0$ changelog entries.  I deleted it just now so that we can test it with no changelog entries.

**Comment by namurphy** [comment]
When I removed the changelog entry, I got this error again:
```
/home/runner/work/PlasmaPy/PlasmaPy/docs/whatsnew/index.rst:11: WARNING: 
toctree contains reference to document 'whatsnew/dev' that doesn't have 
a title: no link will be generated
```
Gonna add the changelog entry back...

File: docs/whatsnew/0.9.0.rst (hunk 4 above)
Comment (rocco8773) [reply docs/whatsnew/0.9.0.rst]: Readying a little more about this, this only works if we use the `:changelog-file:` option.  We still get the an error since we're include a blank file (i.e. a file without a heading) in the `toctree` in `docs/whatsnew/index.rst`.

My suggestion, we keep this file as `dev.rst` with the following content...

```rst
:orphan:

.. changelog::
   :towncrier: ../../

```

And we just delete the `dev` line in `docs/whatsnew/index.rst` during a release and add it back after the release.  We would not need to do anything with `dev.rst` because we've marked it as `:orphan:`.

**Comment by rocco8773** [comment]
> When I removed the changelog entry, I got this error again:
> 
> ```
> /home/runner/work/PlasmaPy/PlasmaPy/docs/whatsnew/index.rst:11: WARNING: 
> toctree contains reference to document 'whatsnew/dev' that doesn't have 
> a title: no link will be generated
> ```
> 
> Gonna add the changelog entry back...

See my above comment... https://github.com/PlasmaPy/PlasmaPy/pull/1623#discussion_r914382006

File: docs/whatsnew/0.9.0.rst (hunk 4 above)
Comment (namurphy) [reply docs/whatsnew/0.9.0.rst]: ✅

**Comment by rocco8773** [comment]
@namurphy Do plan to add the changelog entry back in for this PR?  I think that's all that's missing, so I'll give an approve once it's put back.

**Review by rocco8773 — APPROVED** [review]
